File: auth_server/src/main/java/hu/bme/aut/temalab/authserver/JwkSetEndpoint.java
package hu.bme.aut.temalab.authserver;

import com.nimbusds.jose.jwk.JWKSet;
import com.nimbusds.jose.jwk.RSAKey;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import java.security.KeyPair;
import java.security.interfaces.RSAPublicKey;
import java.util.Map;

@Controller
class JwkSetEndpoint {
    @Autowired
    private RSAKey rsaJWK;

    @GetMapping("/.well-known/jwks.json")
    @ResponseBody
    public Map<String, Object> getKey() {
        RSAKey rsaPublicJWK = rsaJWK.toPublicJWK();
        return new JWKSet(rsaPublicJWK).toJSONObject();
    }
}

File: auth_server/src/main/java/hu/bme/aut/temalab/authserver/JwkSetEndpoint.kt
package hu.bme.aut.temalab.authserver

import com.nimbusds.jose.jwk.JWKSet
import com.nimbusds.jose.jwk.RSAKey
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ResponseBody

@Controller
internal class JwkSetEndpoint {
    @Autowired
    private val rsaJWKAccess: RSAKey? = null

    @get:ResponseBody
    @get:GetMapping("/.well-known/jwks.json")
    val key: Map<String, Any>
        get() {
            val rsaPublicJWK = rsaJWKAccess!!.toPublicJWK()
            return JWKSet(rsaPublicJWK).toJSONObject()
        }
}